src/overlays/actors/ovl_Bg_Hidan_Rock/z_bg_hidan_rock.c

```diff
@@ -47,31 +61,349 @@ static ColliderCylinderInit D_8088BF8C = {
+
+    Actor_ProcessInitChain(&dyna->actor, sInitChain);
+    DynaPolyActor_Init(dyna, DPM_PLAYER);
+
+    this->params = dyna->actor.params & 0xFF;
+    this->unk_169 = 0;
+
+    this->dyna.actor.params = ((dyna->actor.params) >> 8) & 0xFF;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &dyna->actor, &D_8088BF8C);
+
+    if (this->params == 0) {
+        if (Flags_GetSwitch(globalCtx, dyna->actor.params) != 0) {
+            Math_Vec3f_Copy(&this->dyna.actor.home.pos, &D_8088BF60);
+            Math_Vec3f_Copy(&this->dyna.actor.world.pos, &D_8088BF60);
+            this->unk_16A = 0x3C;
+            this->actionFunc = func_8088B5F4;
+        } else {
+            this->actionFunc = func_8088B268;
+        }
+        this->dyna.actor.flags = dyna->actor.flags | 0x30;
+        CollisionHeader_GetVirtual(&D_0600CB80, &col_header);
+    } else {
+        CollisionHeader_GetVirtual(&D_0600DF78, &col_header);
+        this->collider.dim.pos.x = dyna->actor.home.pos.x;
+        this->collider.dim.pos.y = dyna->actor.home.pos.y;
+        this->collider.dim.pos.z = dyna->actor.home.pos.z;
+        this->actionFunc = func_8088B634;
+    }
+
+    this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &dyna->actor, col_header);
+    this->unk_16A = 0;
+}
+
+void BgHidanRock_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgHidanRock* this = THIS;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+    func_800F89E8(&this->unk_170);
+}
+
+void func_8088B24C(BgHidanRock* this) {
+    this->dyna.actor.flags = this->dyna.actor.flags | 0x30;
+    this->actionFunc = func_8088B990;
+}
+
+void func_8088B268(BgHidanRock* this, GlobalContext* globalCtx) {
+    static f32 D_8088BFC0 = 0.0f;
+    f32 sp2C;
+    s32 temp_v1;
+    s32 frame;
+    Player* player = PLAYER;
+
+    if (this->dyna.unk_150 != 0.0f) {
+        if (this->unk_16A == 0) {
+            if (D_8088BFC0 == 0.0f) {
+                if (this->dyna.unk_150 > 0.0f) {
+                    D_8088BFC0 += 0.01f;
+                } else {
+                    D_8088BFC0 -= 0.01f;
+                }
+            }
+
+            this->dyna.actor.speedXZ = this->dyna.actor.speedXZ + 0.05f;
+            this->dyna.actor.speedXZ = (this->dyna.actor.speedXZ > 2.0f) ? 2.0f : this->dyna.actor.speedXZ;
+
+            if (D_8088BFC0 > 0.0f) {
+                temp_v1 = Math_StepToF(&D_8088BFC0, 20.0f, this->dyna.actor.speedXZ);
+            } else {
+                temp_v1 = Math_StepToF(&D_8088BFC0, -20.0f, this->dyna.actor.speedXZ);
+            }
+
+            this->dyna.actor.world.pos.x = (Math_SinS(this->dyna.unk_158) * D_8088BFC0) + this->dyna.actor.home.pos.x;
+            this->dyna.actor.world.pos.z = (Math_CosS(this->dyna.unk_158) * D_8088BFC0) + this->dyna.actor.home.pos.z;
+
+            if (temp_v1 != 0) {
```

Comment:
```suggestion
            if (temp_v1) {
```
`Math_StepToF` returns a boolean.